tests/integration/components/sl-textarea-test.js

```diff
@@ -0,0 +1,305 @@
+        'Textarea is readonly as expected'
+    );
+});
+
+test( '"rows" property is supported', function( assert ) {
+    this.render( hbs`
+        {{#sl-textarea rows="4"}}
+        {{/sl-textarea}}
+    ` );
+
+    assert.strictEqual(
+        this.$( '>:first-child' ).find( 'textarea' ).attr( 'rows' ),
+        '4',
+        'Textarea rows attribute is expected value'
+    );
+});
+
+test( '"helpText" is rendered if populated', function( assert ) {
+    this.set( 'helpText', 'Help Text' );
+    this.render( hbs`
+        {{#sl-textarea helpText=helpText}}
+        {{/sl-textarea}}
+    ` );
+
+    assert.strictEqual(
+        this.$( '>:first-child' ).find( '.help-block' ).prop( 'tagName' ),
+        'P',
+        'Help text block is rendered as a <p>'
+    );
+
+    assert.strictEqual(
+        Ember.$.trim( this.$( '>:first-child' ).find( '.help-block' ).text() ),
+        this.get( 'helpText' ),
+        'Help text block text is expected value'
+    );
+});
+
+test( '"optional" and "required" elements are rendered if populated along with "label" property', function( assert ) {
+    this.render( hbs`
+        {{#sl-textarea label='Test Label' optional=true required=true}}
+        {{/sl-textarea}}
+    ` );
+
+    assert.strictEqual(
+        this.$( '>:first-child' ).find( 'label > .text-info' ).prop( 'tagName' ),
+        'SMALL',
+        "Label's text-info <small> is rendered"
+    );
+
+    assert.strictEqual(
+        this.$( '>:first-child' ).find( 'label > .text-danger' ).prop( 'tagName' ),
+        'SMALL',
+        "Label's text-danger <small> is rendered"
+    );
+});
+
+test(
+    'If "label" property is not populated, "optional" and "required" elements are not rendered even if populated',
+    function( assert ) {
+        this.render( hbs`
+            {{#sl-textarea optional=true required=true}}
+            {{/sl-textarea}}
+        ` );
+
+        assert.strictEqual(
+            this.$( '>:first-child' ).find( 'label > .text-info' ).length,
+            0,
+            "Label's text-info is not rendered"
+        );
+
+        assert.strictEqual(
+            this.$( '>:first-child' ).find( 'label > .text-danger' ).length,
+            0,
+            "Label's text-danger is not rendered"
+        );
+    }
+);
+
+test( 'If "label" property is not populated, label element is not rendered', function( assert ) {
+    this.render( hbs`
+        {{#sl-textarea label=""}}
+        {{/sl-textarea}}
+    ` );
+    assert.strictEqual(
+        Ember.typeOf( this.$( '>:first-child' ).find( 'label' ).prop( 'for' ) ),
+        'undefined',
+        'Label element is not rendered'
+    );
+});
+
+test( 'If "label" property is populated, label element is rendered', function( assert ) {
+    this.set( 'label', 'test' );
+    this.render( hbs`
+        {{#sl-textarea label="test"}}
+        {{/sl-textarea}}
+    ` );
+
+    const label = this.$(
+        'label[for="' + this.$( '>:first-child' ).find( 'textarea' ).prop( 'id' ) + '"]'
```

Comment:
Only look for a `label` element - the value of `for` is tested in the subsequent test.